psi4/driver/driver_cbs.py

```diff
@@ -1518,43 +2075,48 @@ def cbs(func, label, **kwargs):
     for mc in JOBS:
         instructions += """   %12s / %-24s for  %s%s\n""" % \
             (mc['f_wfn'], mc['f_basis'] + " + options"*bool(mc['f_options']),
-             VARH[mc['f_wfn']][mc['f_wfn']], addlremark[ptype])
+             VARH[mc['f_wfn']][mc['f_component']], addlremark[ptype])
 
     #     Remove duplicate modelchem portion listings
     for mc in MODELCHEM:
         dups = -1
         for indx_job, job in enumerate(JOBS):
             if (job['f_wfn'] == mc['f_wfn']) and (job['f_basis'] == mc['f_basis']) and \
-               (job['f_options'] == mc['f_options']):
+               (job['f_options'] == mc['f_options']) and \
+               (job['f_options'] != False or ptype == 'energy'):
                 dups += 1
                 if dups >= 1:
                     del JOBS[indx_job]
 
-    #     Remove chemically subsumed modelchem portion listings
-    if ptype == 'energy':
-        for mc in MODELCHEM:
-            for wfn in VARH[mc['f_wfn']]:
+    #     Remove chemically subsumed modelchem portion listings for energies and DFT
+        if ptype == "energy" or mc['f_wfn'] in functionals and mc['f_wfn'] not in ["scf", "hf"]:
```

Comment:
Are the "Remove duplicate" and "Remove chemically subsumed" loops combined for a reason here or just a refactoring opportunity? Might have trouble with intermediate step in ddd. https://github.com/psi4/psi4/pull/1351/files#diff-9e10a6f2a977cb3eb6f9c34dea0a6d54110f18bbe841bf9f116107485612e26fR1269-R1272

Reply:
I think they do different things. Note the `ptype == "energy"` in the second loop; while we have the HF energy when we do MP2, we *don't* have it's gradient, even though the energy is subsumed. I think there might be a way to combine these two. 

In my view, the best way to clarify this is to define the whole VARH for (analytical) E/G/H.